### PR TITLE
Fixed G-code

### PR DESCRIPTION
**Pull request by TannerReynolds**
- Changed from `data` to `programming` -> CNC Programmers create gcode files that instruct machines where to move and what to do
- Gave a color (#ff4d6b)
- Added a commonly used file type for CNC programming (.cnc)

## Description
G-code files are instructions written or generated by CNC/machine programmers to instruct machines where to move and what to do (Spindle speeds, move speed, feedrate, tooling, etc.) on a line-by-line basis. As a CNC/Machine programmer, I can tell you that they are not data files, just like any other high level language, they are written instructions to be interpreted by the machine. I have also included a file type that is commonly used for CNC programming.

## Checklist:

- [x] **I am associating a language with a new file extension.** (Not new to GCode programming, but new to Github)
  - [?] The new extension is used in hundreds of repositories on GitHub.com Because it is currently not a programming language w/color, I cannot see the repository count. I do see thousands of gcode files with this extension, though.
    - Search results for each extension:
      
      - https://github.com/search?q=extension%3Acnc+NOT+nothack&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/TannerReynolds/GHL
    - Sample license(s): No License for sample
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension. (While syntax is mostly the same, different machinery use different syntax, this language is a hard one to detect using heuristics)

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [] I have updated the heuristics to distinguish my language from others using the same extension.

- [x] **I am fixing a misclassified language**
  - [x] I have included a new sample for the misclassified language:
    - Sample source(s):
      - https://github.com/TannerReynolds/GHL
    - Sample license(s): No license for sample
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/
  
- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1471,11 +1471,13 @@ Frege:
...
   language_id: 116
 G-code:
-  type: data
+  type: programming
+  color: "#ff4d6b"
   extensions:
   - ".g"
   - ".gco"
   - ".gcode"
+  - ".mpf"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Please fill the PR template appropriately for that new extension.

**Comment by TannerReynolds** [comment]
@pchaigno I went ahead and fixed the commits/PR comment

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -1471,11 +1471,13 @@ Frege:
...
   language_id: 116
 G-code:
-  type: data
+  type: programming
+  color: "#ff4d6b"
   extensions:
   - ".g"
   - ".gco"
   - ".gcode"
+  - ".cnc"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Except for the first extension, extensions should be sorted alphabetically.

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -1471,11 +1471,13 @@ Frege:
   ace_mode: haskell
   language_id: 116
 G-code:
-  type: data
+  type: programming
+  color: "#ff4d6b"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Please see the Travis-CI build failure. That color is too close to NetLogo's.

File: lib/linguist/languages.yml (hunk 2 above)
Comment (pchaigno) [inline lib/linguist/languages.yml]: For other reviewers: using Harvester, I counted 341 repositories by 240 users over a total of 2389 files checked.

**Comment by TannerReynolds** [comment]
> Are these files written by hand by humans or through a GUI/higher-level language?
@pchaigno 
It depends. Small-medium sized manufacturers, hobbyists, people who prefer it, and even some bigger manufacturers have people who write all the code themselves. Usually manufacturers that require a large volume of code to be written daily, and that have lots of money, will opt-out for some sort of generator, that is easier to teach, but far more expensive to use. Otherwise people generally tend to prefer hand writing the code, due to added control over the machines. I for one know how to do both, and decide on whether I'm going to code it myself, or use CAM to do it, on a use-by-use basis.

**Pull request merged**